app/src/renderer/components/staking/ModalStake.vue

```diff
@@ -0,0 +1,112 @@
+<template lang="pug">
+  .modal-stake
+    .stake-header
+      img.icon(class='stake-atom' src="~assets/images/cosmos-logo.png")
+      span.tm-modal-title Stake
+
+    tm-form-group.stake-form-group(
+      :error='$v.amount.$invalid'
+      field-id='amount'
+      field-label='Amount'
+    )
+      tm-field#amount(
+        type="number"
+        v-model="amount")
+      tm-form-msg(name='Amount' type='between' :min='1' :max='maximum'
+        v-if='$v.amount.$invalid')
+
+    tm-form-group.stake-form-group(
+      field-id='to' field-label='To')
+      tm-field#to(v-model="to")
+
+    tm-form-group.stake-form-group(
+      field-id='from' field-label='From')
+      tm-field#from(
+        type="select"
+        :options="fromOptions"
+        v-model="fromIndex"
+      )
+
+    .stake-footer
+      tm-btn(
+        @click.native="onStake"
+        color="primary"
+        :disabled="$v.amount.$invalid"
+        size="lg"
+        value="Stake"
+      )
+</template>
+
+<script>
+import { between } from "vuelidate/lib/validators"
+import Modal from "common/TmModal"
+import { TmBtn, TmField, TmFormGroup, TmFormMsg } from "@tendermint/ui"
+
+export default {
+  props: [`fromOptions`, `maximum`, `to`],
+  components: {
+    Modal,
+    TmBtn,
+    TmField,
+    TmFormGroup,
+    TmFormMsg
+  },
+  data: () => ({
+    amount: 1,
+    fromIndex: 0
+  }),
+  validations() {
+    return {
+      amount: {
+        between: between(1, this.maximum)
```

Comment:
Imo it should be `between(0.0000000001, this.maximum)` we should allow for decimals as well

Reply:
Apparently we don't allow decimals. Decimals will be `sub-atoms`.